physics/inertia_tensor_body.hpp

```diff
@@ -0,0 +1,123 @@
+#include "geometry/rotation.hpp"
+
+namespace principia {
+namespace physics {
+namespace internal_inertia_tensor {
+
+using geometry::Barycentre;
+using geometry::Bivector;
+using geometry::Identity;
+using geometry::Rotation;
+using geometry::SymmetricProduct;
+
+template<typename Frame>
+InertiaTensor<Frame>::InertiaTensor(
+    Mass const& mass,
+    R3x3Matrix<MomentOfInertia> const& coordinates,
+    Position<Frame> const& centre_of_mass)
+    : InertiaTensor(mass,
+                    MakeSymmetricBilinearForm(coordinates),
+                    centre_of_mass) {}
+
+template<typename Frame>
+Position<Frame> const& InertiaTensor<Frame>::centre_of_mass() const {
+  return centre_of_mass_;
+}
+
+template<typename Frame>
+template<typename ToFrame>
+InertiaTensor<ToFrame> InertiaTensor<Frame>::Transform(
+    RigidTransformation<Frame, ToFrame> const& transformation) const {
+  static Identity<Frame, ToFrame> const identity{};
+
+  auto const& from_origin = transformation.from_origin();
+  auto const& to_origin = transformation.to_origin();
+  auto const& linear_map = transformation.linear_map();
+
+  Displacement<ToFrame> const translation =
+      identity(from_origin - Frame::origin) - (to_origin - ToFrame::origin);
+  SymmetricBilinearForm<MomentOfInertia, ToFrame> const translated_form =
+      linear_map(form_) + mass_ * SymmetricProduct(translation, translation);
+  Position<ToFrame> const transformed_centre_of_mass =
+      transformation(centre_of_mass_);
+  return InertiaTensor<ToFrame>(mass_,
+                                translated_form,
+                                transformed_centre_of_mass);
+}
+
+template<typename Frame>
+template<typename PrincipalAxesFrame>
+typename InertiaTensor<Frame>::PrincipalAxes<PrincipalAxesFrame>
+InertiaTensor<Frame>::Diagonalize() const {
+  struct IntermediateFrame {};
+
+  auto const eigensystem = form_.Diagonalize<IntermediateFrame>();
+
+  // The eigenvalues are {Σx², Σy², Σz²} in increasing order.
+  R3x3Matrix<MomentOfInertia> const eigensystem_form_coordinates =
+      eigensystem.form.coordinates();
+
+  // The moment of inertia in increasing order are
+  // {Σ(x² + y²), Σ(x² + z²), Σ(y² + z²)}, which does *not* correspond to the
+  // ordered eigenvalues of the "classical" inertia tensor, which are
+  // {Σ(y² + z²), Σ(x² + z²), Σ(x² + y²)}.  We address this by multiplying the
+  // rotation by another one that swaps x and z.
+  R3Element<MomentOfInertia> const moments_of_inertia(
+      eigensystem_form_coordinates(0, 0) + eigensystem_form_coordinates(1, 1),
+      eigensystem_form_coordinates(2, 2) + eigensystem_form_coordinates(0, 0),
+      eigensystem_form_coordinates(1, 1) + eigensystem_form_coordinates(2, 2));
+
+  Rotation<IntermediateFrame, PrincipalAxesFrame> const swap_x_and_z(
+      Bivector<double, IntermediateFrame>({0, 0, 1}),
+      Bivector<double, IntermediateFrame>({0, 1, 0}),
+      Bivector<double, IntermediateFrame>({-1, 0, 0}));
```

Comment:
Name the first two bivectors `z` and `y`, and pass `Commutator(z, y)` as the third argument so that correctness is obvious.

Reply:
Done.